liquibase-core/src/test/groovy/liquibase/LiquibaseTest.groovy

```diff
@@ -491,6 +503,32 @@ class LiquibaseTest extends Specification {
         assertSqlOutputAppliesTags(writer.toString(), "1.1");
     }
 
+    def validateContextLabelEntryHasNotBeenAddedPreviously() {
```

Comment:
Thanks for adding these tests, @MalloD12! Should we have a test for the case where there are no labels and no contexts? I'm unclear as to what should happen in that case.

Reply:
Hi Erz, I think that scenario is not needed because if I'm not wrong if we provide a changeset with context/label as empty strings will work in the same way as with any string values because it will then create a key to validate whether changeset has been already processed or not without the need to acquire a lock. Here is part of the logic I'm talking about:
```
 String cacheKey = contexts +"/"+ labelExpression;
 if (!this.upToDateFastCheck.containsKey(cacheKey)) {
<some code>
}
```
Now, if you meant without providing context/label properties in the changeset then this method would not be called because this is going through the path of this update:
`public void update(Contexts contexts, LabelExpression labelExpression, boolean checkLiquibaseTables)` 

Thanks,
Daniel.

Reply:
@MalloD12, hi - It was the second case I was referring to. Thanks for explaining that case will not go through the same code.